src/topology.cpp

```diff
@@ -1,57 +1,18 @@
 #include <lo2s/topology.hpp>
+#include <lo2s/util.hpp>
 
 namespace lo2s
 {
 const std::filesystem::path Topology::base_path = "/sys/devices/system/cpu";
 
-namespace detail
-{
-static auto parse_list(std::string list) -> std::set<uint32_t>
-{
-    std::stringstream s;
-    s << list;
-
-    std::set<uint32_t> res;
-
-    std::string part;
-    while (std::getline(s, part, ','))
-    {
-        auto pos = part.find('-');
-        if (pos != std::string::npos)
-        {
-            // is a range
-            uint32_t from = std::stoi(part.substr(0, pos));
-            uint32_t to = std::stoi(part.substr(pos + 1));
-
-            for (auto i = from; i <= to; ++i)
-                res.insert(i);
-        }
-        else
-        {
-            // single value
-            res.insert(std::stoi(part));
-        }
-    }
-
-    return res;
-}
-} // namespace detail
-
 void Topology::read_proc()
 {
     std::string online_list;
-    std::string present_list;
-
-    {
-        std::ifstream cpu_online(base_path / "online");
-        std::getline(cpu_online, online_list);
 
-        std::ifstream cpu_present(base_path / "present");
-        std::getline(cpu_present, present_list);
-    }
+    std::ifstream cpu_online(base_path / "online");
+    std::getline(cpu_online, online_list);
 
-    auto online = detail::parse_list(online_list);
-    auto present = detail::parse_list(present_list);
+    auto online = parse_list(online_list);
```

Comment:
add another function `parse_list_from_file` which handles the file reading as well